Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -2421,10 +2421,9 @@ double HTMLInputElement::step_base() const
 
     // 2. If the element has a value content attribute, and the result of applying the algorithm to convert a string to a number to the value of
     // the value content attribute is not an error, then return that result.
-    // AD-HOC: https://github.com/whatwg/html/issues/11097 Skipping this step seems to be necessary in order to get the
-    //         behavior that's actually expected — and necessary in order get the tests to pass (they otherwise fail).
-    // if (auto value = convert_string_to_number(this->value()); value.has_value())
-    // return *value;
+    if (auto value_content_attribute = get_attribute(HTML::AttributeNames::value); value_content_attribute.has_value())
+        if (auto value_content_attribute_as_number = convert_string_to_number(value_content_attribute.value()); value_content_attribute_as_number.has_value())
+            return value_content_attribute_as_number.value();
```

Comment:
This outer `if` is multi-line and thus should have braces:
```c++
if (auto value_content_attribute = get_attribute(HTML::AttributeNames::value); value_content_attribute.has_value()) {
    if (auto value_content_attribute_as_number = convert_string_to_number(value_content_attribute.value()); value_content_attribute_as_number.has_value())
        return value_content_attribute_as_number.value();}
}
```

(Also nit: `value_content_attribute` and `value_content_attribute_as_number` sound super verbose to me, and causes these lines to be quite long. IMO just `value` and `value_as_number` seems fine).